src/core/loss_detection.c

```diff
@@ -1496,6 +1498,69 @@ QuicLossDetectionProcessAckBlocks(
     }
 
     if (NewLargestAck) {
+        if (Connection->Settings.EcnEnabled && Path->EcnValidationState != ECN_VALIDATION_FAILED) {
+            //
+            // Per RFC 9000, we validate ECN counts from received ACK frames
+            // when the largest acked packet number increases.
+            //
+            BOOLEAN EcnValidated = TRUE;
+            int64_t EctCeDeltaSum = 0;
+            if (Ecn != NULL) {
+                EctCeDeltaSum +=
+                    Ecn->CE_Count - Connection->EcnCeCounters[EncryptLevel];
+                EctCeDeltaSum +=
+                    Ecn->ECT_0_Count - Connection->EcnEctCounters[EncryptLevel];
```

Comment:
Can we add some debug asserts to make sure the subtraction doesn't ever underflow?

Reply:
It is a real possibility. That's why we need to validate the counts.